src/full_info.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement.  This, along with the Licenses can be
+// found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use public_info::PublicInfo;
+use rust_sodium::crypto::{box_, sign};
+use xor_name::XorName;
+
+/// Network identity component containing name, and public and private keys.
```

Comment:
We use "secret" now rather than "private".  Also, we're not keen on calling this an identity.  I guess I'd change this comment to something like:
```
/// A container for a peer's cryptographic keys, and in the case of a node, its age too.  The public
/// signing key defines the peer's name on the network.
```